templates/terraform/resource.erb

```diff
@@ -85,18 +85,16 @@ func resource<%= resource_name -%>Create(d *schema.ResourceData, meta interface{
   }
   <% end -%>
 
+	obj := make(map[string]interface{})
 <% settable_properties.each do |prop| -%>
 	<%= prop.api_name -%>Prop, err := expand<%= resource_name -%><%= titlelize_property(prop) -%>(d.Get("<%= Google::StringUtils.underscore(prop.name) -%>"), d, config)
 	if err != nil {
 		return err
+	} else if v, ok := d.GetOkExists("<%= Google::StringUtils.underscore(prop.name) -%>"); ok || !reflect.DeepEqual(v, <%= prop.api_name -%>Prop) {
```

Comment:
Let me make sure I understand the second half of the statement. This seems to handle the case where the value was not set (therefore its value will be the zero value) and its value (the zero value) is not equal to what expand produced. So this is for if it wasn't set, but we still do stuff in expand for it? So things like region would be the example case, right? Is there another type of field that would do it? Either way, mind adding a comment (in ruby blocks is fine) explaining this case?

Reply:
You got it, and you're 100% right.  I will add a comment as soon as I am convinced, myself, that it works.  :)